sparkle/src/components/ContextItem.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ComponentType, ReactNode } from "react";
+
+import { classNames } from "@sparkle/lib/utils";
+
+import { Icon } from "./Icon";
+
+type ContextItemProps = {
+  title: string;
+  visual: ReactNode;
+  action?: ReactNode;
+  children?: ReactNode;
+  hasSeparator?: boolean;
+};
+
+export function ContextItem({
+  title,
+  visual,
+  action,
+  children,
+  hasSeparator = true,
+}: ContextItemProps) {
+  return (
+    <div
+      className={classNames(
+        hasSeparator ? "s-border-b s-border-structure-200" : "",
+        "s-flex s-w-full s-flex-col"
+      )}
+      aria-label={title}
+    >
+      <div className="s-flex s-flex-row s-gap-3 s-pb-5 s-pt-3">
+        <div className="s-flex s-pt-1.5">{visual}</div>
+        <div className="s-flex s-grow s-flex-col">
+          <div className="s-text-normal s-flex s-h-9 s-flex-col s-justify-center s-font-semibold">
+            {title}
+          </div>
+          <div className="-s-mt-1">{children}</div>
+        </div>
+        <div>{action}</div>
+      </div>
+    </div>
+  );
+}
+
+interface ContextItemListProps {
+  children: ReactNode;
+  className?: string;
+}
+
+ContextItem.List = function ({ children, className }: ContextItemListProps) {
+  // Ensure all children are of type ContextItem
+  React.Children.forEach(children, (child) => {
```

Comment:
I'm a bit worried that this isn't enforce by typescript. I don't think we really typecheck the children. Is there a strong reason not to accept an array of props and have the `ContextItem` be created from the component instead ?

Reply:
It activate or deactivate "hasSeparator' of the item to create a list with separator.
So it does need to make sure it's a ContextItem (to know it has a hasSeparator.